### PR TITLE
CI: switch to Lit pool

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

changing the pool

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [ ] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .azure/gpu-tests.yml [hunk 1]
```diff
@@ -18,12 +18,12 @@ jobs:
...
     cancelTimeoutInMinutes: "2"
 
-    pool: azure-gpus-spot
+    pool: lit-rtx-3090
 
     container:
       image: "pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.11-cuda11.3.1"
       #endpoint: azureContainerRegistryConnection
-      options: "--runtime=nvidia -e NVIDIA_VISIBLE_DEVICES=all --shm-size=32g --name ci-container -v /usr/bin/docker:/tmp/docker:ro"
+      options: "--gpus=all --shm-size=16g --name ci-container -v /usr/bin/docker:/tmp/docker:ro"
```
Comment (otaj) [inline .azure/gpu-tests.yml:26]: isn't it also needed to have a step `set visible devices` as in main repo?

**Review by otaj — APPROVED** [review]
Thank you! Does it need some action on Azure side afterwards? or Could I have done that myself? :zap:

File: .azure/gpu-tests.yml (hunk 1 above)
Comment (Borda) [reply .azure/gpu-tests.yml:28]: good catch

**Review by akihironitta — APPROVED** [review]

**Comment by Borda** [comment]
assuming to merge this one as the failing test seems to be the same as on the master
so let's move to resolve this test to another PR :otter:

**Pull request merged**